qcengine/util.py

```diff
@@ -220,22 +266,52 @@ def popen(
         # Allow using CTRL_C_EVENT / CTRL_BREAK_EVENT
         popen_kwargs["creationflags"] = subprocess.CREATE_NEW_PROCESS_GROUP
 
+    # Route the standard error and output
     popen_kwargs["stdout"] = subprocess.PIPE
     popen_kwargs["stderr"] = subprocess.PIPE
+
+    # Prepare StringIO objects to store the stdout and stderr
+    stdout = io.BytesIO()
+    stderr = io.BytesIO()
+
+    # Launch the process
     LOGGER.info("Popen", args, popen_kwargs)
+
+    # Ready the output
     ret = {"proc": subprocess.Popen(args, **popen_kwargs)}
+
+    # Spawn threads that will read from the stderr/stdout
+    #  The PIPE uses a buffer with finite capacity. The underlying
+    #  process will stall if it is unable to write to the buffer
+    #  because the buffer is full. These threads continuously read
+    #  from the buffers to ensure that they do not fill.
+    #
```

Comment:
How much is that output BTW? Curious how you hit this cap, we often dumps 20-30MB log files here and haven't had an issue yet.

Reply:
The problem was with the `stderr`, which had maybe a few KBs of data. The MPI tasks I was using print a stack trace from each MPI rank to `stderr`, which probably explains why I was seeing this problem for the first time. This change (also in #188) adds the same kind of buffer unpacking that was already used for stdout.